plugins/modules/net_tools/dnsimple_info.py

```diff
@@ -0,0 +1,179 @@
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: dnsimple_info
+
+short_description: Pull basic info from DNSimple API
+
+version_added: "1.0.0"
+
+description: This is my longer description explaining my test info module.
+
+options:
+    name:
+        description: The domain name to retrieve info from, will return all associated records
+        required: false
+        type: str
+
+    account_id:
+        description: The account ID to query
+        required: true
+        type: str
+
+    api_key:
+        description: The API key to use
+        required: true
+        type: str
+
+    record:
+        description: The record to find
+        required: false
+        type: str
+
+    sandbox:
+        description: whether or not to use sandbox environment
+        required: false
+        default: false
+        type: bool
+
+author:
+    - Your Name (@yourGitHubHandle)
+'''
+
+EXAMPLES = r'''
+- name: Get all domains from an account
+  dnsimple_info:
+    account_id: "1234"
+    api_key: "1234"
+
+- name: Get all records from a domain
+  dnsimple_info:
+    name: "example.com"
+    account_id: "1234"
+    api_key: "1234"
+
+- name: Get all info from a matching record
+  dnsimple_info:
+    name: "example.com"
+    record: "subdomain"
+    account_id: "1234"
+    api_key: "1234"
+'''
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    from requests import Request, Session
+except ImportError:
+    HAS_ANOTHER_LIBRARY = False
+    ANOTHER_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_ANOTHER_LIBRARY = True
+
+
+def build_url(account, key, is_sandbox):
+    headers = {'Accept': 'application/json',
+               'Authorization': 'Bearer ' + key}
+    url = 'https://api{sandbox}.dnsimple.com/'.format(
+        sandbox=".sandbox" if is_sandbox else "") + 'v2/' + account
+    req = Request(url=url, headers=headers)
+    prepped_request = req.prepare()
+    return prepped_request
+
+
+def iterate_data(request_object):
+    data = []
+    response = Session().send(request_object)
+    if response.json()["pagination"]["total_pages"]:
+        pages = response.json()["pagination"]["total_pages"]
+        for p in range(pages):
+            p = p + 1
+            request_object.url = request_object.url + '&page=' + str(p)
+            new_results = Session().send(request_object)
+            if new_results.json()["data"][0]:
+                data.append(new_results.json()["data"][0])
+        return(data)
+    else:
+        return response.json()
+
+
+def record_info(account, key, domain, record, req_obj):
+    req_obj.url, req_obj.method = req_obj.url + '/zones/' + domain + '/records?name=' + record, 'GET'
+    return iterate_data(req_obj)
+
+
+def domain_info(account, key, domain, req_obj):
+    req_obj.url, req_obj.method = req_obj.url + '/zones/' + domain + '/records?per_page=1', 'GET'
```

Comment:
Is it a good idea to restrict output to one element per page, and then iterate over all pages? That sounds pretty inefficient to me.

Reply:
It is inefficient, and something I am going to revise, but ran into some issues building my data when altering items per page. Is it possible for me to add this as a TODO within the script? I really appreciate all your input and help

Reply:
This is something you should really fix for the first version to be released. Pagination with page size 1 is ok for testing, but not for a release.